### PR TITLE
Add conversation message metadata prop

### PR DESCRIPTION
**Pull request by petersalas**
This allows hosting/persistence environments such as Fixie to read/write additional message-level metadata.

### CODE REVIEW

File: packages/ai-jsx/src/core/conversation.tsx [hunk 1]
```diff
@@ -16,7 +17,7 @@ import _ from 'lodash';
  *    </ChatCompletion>
  * ```
  */
-export function SystemMessage({ children }: { children: Node }) {
+export function SystemMessage({ children }: { children: Node; metadata?: Record<string, Jsonifiable> }) {
```
Comment (farzadab) [inline packages/ai-jsx/src/core/conversation.tsx:20]: Noob TypeScript question: so you can define something in the type and not even use it? That's cool and now I realize I've needed this so much and never realized! 😅

**Review by farzadab — APPROVED** [review]
LGTM, thanks!

File: packages/ai-jsx/src/core/conversation.tsx (hunk 1 above)
Comment (petersalas) [reply packages/ai-jsx/src/core/conversation.tsx:20]: The key is that the identifiers on the left side of the `:` are _variable_ declarations -- the linter will get upset at you if you declare unused variables but nothing will get upset about unused fields on a type.

File: packages/fixie-sdk/src/request-wrapper.tsx [hunk 2]
```diff
@@ -39,6 +39,7 @@ export function FixieRequestWrapper({
                 {{
                   kind: 'text',
                   content: message.element,
+                  metadata: message.element.props.metadata,
```
[the remark this answers is not in the document]
Comment (petersalas) [reply packages/fixie-sdk/src/request-wrapper.tsx:42]: Good callout -- yes, `JSON.stringify` strips any fields with `undefined` as a value.

**Comment by NickHeiner** [comment]
@CodiumAI-Agent /review

**Review by NickHeiner — APPROVED** [review]

**Comment by farzadab** [comment]
@petersalas will you be merging this?
My next PR is based on this and I wanna reduce conflicts.

**Pull request merged**